### PR TITLE
GODRIVER-1962 unwrap errors in WithTransaction

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: mongo/session.go [hunk 1]
```diff
@@ -193,9 +194,12 @@ func (s *sessionImpl) WithTransaction(ctx context.Context, fn func(sessCtx Sessi
 			default:
 			}
 
-			if cerr, ok := err.(CommandError); ok {
-				if cerr.HasErrorLabel(driver.TransientTransactionError) {
-					continue
+			currErr := err
+			for ; currErr != nil; currErr = unwrap(currErr) {
```
Comment (benjirewis) [inline mongo/session.go:198]: Is this the same as `errors.As(...)`, just for < Go 1.13?

**Review by benjirewis — APPROVED** [review]
LGTM just a question.

File: mongo/session.go (hunk 1 above)
Comment (iwysiu) [reply mongo/session.go:198]: Yup! This is the workaround we add for the error helpers. It's basically what errors.As does

File: mongo/session.go [hunk 2]
```diff
@@ -193,9 +194,12 @@ func (s *sessionImpl) WithTransaction(ctx context.Context, fn func(sessCtx Sessi
...
+				if cerr, ok := currErr.(CommandError); ok {
```
Comment (divjotarora) [inline mongo/session.go:199]: nit: You can reduce the nesting level by merging this and the check below into `ok && cerr.HasErrorLabel(...)`.

File: mongo/session.go (hunk 1 above)
Comment (divjotarora) [inline mongo/session.go:198]: Can we move this into a separate helper function? That way, we wouldn't be adding an additional loop into this function and would not need the new `RetryLoop` label, which is making this code harder to read IMO.

Comment (iwysiu) [reply mongo/session.go]: done

File: mongo/session.go (hunk 2 above)
Comment (iwysiu) [reply mongo/session.go]: code removed

File: mongo/with_transactions_test.go [hunk 3]
```diff
@@ -381,6 +393,24 @@ func TestConvenientTransactions(t *testing.T) {
...
+	t.Run("wrapped transient transaction error retried", func(t *testing.T) {
+		sess, err := client.StartSession()
+		assert.Nil(t, err, "StartSession error: %v", err)
+		defer sess.EndSession(context.Background())
+
+		returnError := true
+		res, err := sess.WithTransaction(context.Background(), func(sessCtx SessionContext) (interface{}, error) {
+			if returnError {
+				returnError = false
+				return nil, wrappedError{CommandError{Name: "test Error", Labels: []string{driver.TransientTransactionError}}}
```
Comment (divjotarora) [inline mongo/with_transactions_test.go:405]: nit: split this into multiple lines for readability.

File: mongo/with_transactions_test.go [hunk 4]
```diff
@@ -381,6 +393,24 @@ func TestConvenientTransactions(t *testing.T) {
 		// Assert that transaction is canceled within 500ms and not 2 seconds.
 		assert.Soon(t, callback, 500*time.Millisecond)
 	})
+	t.Run("wrapped transient transaction error retried", func(t *testing.T) {
+		sess, err := client.StartSession()
+		assert.Nil(t, err, "StartSession error: %v", err)
+		defer sess.EndSession(context.Background())
+
+		returnError := true
```
Comment (divjotarora) [inline mongo/with_transactions_test.go:401]: Can you add a comment describing what this is doing? It took me a couple reads to realize that `returnError` is being used to track whether the callback is executing only once or is being retried.

**Review by divjotarora — APPROVED** [review]
Minor nits, but LGTM.

File: mongo/with_transactions_test.go (hunk 3 above)
Comment (iwysiu) [reply mongo/with_transactions_test.go]: done

**Pull request merged**